core/src/main/java/org/datasyslab/geospark/geometryObjects/Circle.java

```diff
@@ -101,59 +105,160 @@ public void setRadius(Double givenRadius) {
 		this.MBR=new Envelope(this.centerPoint.x-this.radius, this.centerPoint.x+this.radius, this.centerPoint.y-this.radius, this.centerPoint.y+this.radius);
 	}
 	
-	/* (non-Javadoc)
-	 * @see com.vividsolutions.jts.geom.Geometry#covers(com.vividsolutions.jts.geom.Geometry)
-	 */
-	@Override
-	public boolean covers(Geometry g) {
-		// short-circuit test
-		if (! getEnvelopeInternal().covers(g.getEnvelopeInternal()))
-			return false;
-		// optimization for rectangle arguments
-		if (isRectangle()) {
-			// since we have already tested that the test envelope is covered
-			return true;
-		}
-		double x1,x2,y1,y2;
-		x1=g.getEnvelopeInternal().getMinX();
-		x2=g.getEnvelopeInternal().getMaxX();
-		y1=g.getEnvelopeInternal().getMinY();
-		y2=g.getEnvelopeInternal().getMaxY();	
-		return covers(x1,y1)&&covers(x1,y2)&&covers(x2,y2)&&covers(x2,y1);
-	}
-	
-	/* (non-Javadoc)
-	 * @see com.vividsolutions.jts.geom.Geometry#intersects(com.vividsolutions.jts.geom.Geometry)
-	 */
-	@Override
-	public boolean intersects(Geometry g) {
-		// short-circuit test
-		if (! getEnvelopeInternal().covers(g.getEnvelopeInternal()))
-			return false;
-		// optimization for rectangle arguments
-		if (isRectangle()) {
-			// since we have already tested that the test envelope is covered
-			return true;
-		}
-		double x1,x2,y1,y2;
-		x1=g.getEnvelopeInternal().getMinX();
-		x2=g.getEnvelopeInternal().getMaxX();
-		y1=g.getEnvelopeInternal().getMinY();
-		y2=g.getEnvelopeInternal().getMaxY();	
-		return covers(x1,y1)||covers(x1,y2)||covers(x2,y2)||covers(x2,y1);
-	}
-	
-	/**
-	 * Covers.
-	 *
-	 * @param x the x
-	 * @param y the y
-	 * @return true, if successful
-	 */
-	public boolean covers(double x, double y) {
-		double distance = Math.sqrt((x-this.centerPoint.x)*(x-this.centerPoint.x)+(y-this.centerPoint.y)*(y-this.centerPoint.y));
-		return distance<=this.radius?true:false;
-	}
+    /* (non-Javadoc)
+     * @see com.vividsolutions.jts.geom.Geometry#covers(com.vividsolutions.jts.geom.Geometry)
+     */
+    @Override
+    public boolean covers(Geometry other) {
+        // short-circuit test
+        Envelope otherEnvelope = other.getEnvelopeInternal();
+        if (! getEnvelopeInternal().covers(otherEnvelope)) {
+            return false;
+        }
+
+        if (other instanceof Point) {
+            return covers((Point) other);
+        }
+
+        if (other instanceof LineString) {
+            return covers((LineString) other);
+        }
+
+        if (other instanceof Polygon) {
+            return covers(((Polygon) other).getExteriorRing());
+        }
+
+        if (other instanceof GeometryCollection) {
```

Comment:
@mbasmanova Hi Masha, since you have added the support of GeometryCollection, can you please also add the support for MultiPoint, MultiLineString, MultiPolygon (see https://en.wikipedia.org/wiki/Well-known_text)? MultiPolygon is popular I think. Thank you!